astro_2.0/features/pages/nestedDaoPagesContent/CreateGovernanceTokenPageContent/components/steps/DelegateVoting/DelegateVoting.tsx

```diff
@@ -0,0 +1,200 @@
+      'ft_metadata',
+    ]) as CustomContract;
+
+    const stContract = nearService.getContract(
+      nearService.getStackingContract(dao.name),
+      ['ft_balance_of']
+    ) as CustomContract;
+
+    const [meta, balance] = await Promise.all([
+      ftContract.ft_metadata(),
+      stContract.ft_balance_of({ account_id: accountId }),
+    ]);
+
+    setTokenDetails({
+      balance: Number(formatYoktoValue(balance, meta.decimals)),
+      symbol: meta.symbol,
+      decimals: meta.decimals,
+    });
+  }, [nearService, accountId, dao]);
+
+  const methods = useForm<Form>({
+    mode: 'onChange',
+    defaultValues: {
+      accounts: dao.groups
+        .filter(group => group.slug === 'council')
+        .reduce<AccountAmountDetails[]>((res, group) => {
+          const members = group.members.map(member => ({
+            name: member,
+            amount: 0,
+          }));
+
+          res.push(...members);
+
+          return res;
+        }, []),
+    },
+    resolver: yupResolver(
+      yup.object().shape({
+        accounts: yup.array().of(
+          yup.object().shape({
+            name: yup
+              .string()
+              .test({
+                name: 'notValidNearAccount',
+                exclusive: true,
+                message: 'Only valid near accounts are allowed',
+                test: async value => validateUserAccount(value, nearService),
+              })
+              .required('Required'),
+            amount: yup
+              .number()
+              .min(0, 'Amount must be a positive number.')
+              .typeError('Amount must be a valid number.'),
+          })
+        ),
+      })
+    ),
+  });
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isValid },
+  } = methods;
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'accounts',
+  });
+
+  const submitHandler = async (data: Form) => {
+    if (!nearService || !tokenDetails) {
+      return;
+    }
+
+    // do delegate
+    await nearService.delegateVoting(
+      nearService.getStackingContract(dao.name),
+      data.accounts.map(item => ({
+        ...item,
+        amount: new Decimal(item.amount)
+          .mul(10 ** tokenDetails.decimals)
+          .toFixed(),
+      }))
+    );
+
+    // toggle onsubmit
+    onSubmit();
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <form
+        className={styles.root}
+        noValidate
+        onSubmit={handleSubmit(submitHandler)}
+      >
+        <div className={styles.content}>
+          <div className={styles.header}>
+            <h2>Delegate Voting</h2>
+          </div>
+          <div className={styles.body}>
+            <div className={styles.rows}>
+              {fields.map((item, index) => {
+                return (
+                  <AccountAmount
+                    key={item.name}
+                    item={item}
+                    index={index}
+                    onRemove={() => remove(index)}
```

Comment:
I consider it would be more effective to pass the remove method inside the AccountAmount component and call it there with the index. Otherwise we create a new method with new reference each time.

Reply:
fixed